### ISSUE TITLE
No active plays

### ISSUE THREAD

**Issue by kapiszon53** [alebupal/tvspy#31]
Hi, first of all thanks for developing this, I was waiting long time for a good statistics app.

My issue is that the app is not showing active plays. Everything else seems to be working fine. When I change the channels, this appears for a second and then disappears, showing "No active plays" despite playing a channel:
![Zrzut ekranu 2024-11-13 144436](https://github.com/user-attachments/assets/b842eb1b-f854-4620-a90b-50001e389345)

Thank you!

**Comment by alebupal** [comment]
hi!
In the container log does it tell you that it has connected to the websocket?
![image](https://github.com/user-attachments/assets/0b4661d5-b523-471c-aeee-71fe01f55f29)

**Comment by kapiszon53** [comment]
Yes it does
![image](https://github.com/user-attachments/assets/6eb6e2e5-e94e-420e-967c-3aca92a5c0bb)

**Comment by alebupal** [comment]
the app and tvheadend are on the same machine? I mean if they have the same IP and different ports

**Comment by kapiszon53** [comment]
They are on the same host within different docker containers

**Comment by alebupal** [comment]
Can you pass me what the backend receives from the websocket?
I'll write you the steps:

1. `docker exec -i -t tvspy /bin/sh` -> tvspy is the name of your container
2. `vi /app/backend/src/routes/subcriptions.js`
3. Press enter to be able to write
4. On line 34 ([https://github.com/alebupal/tvspy/blob/53b7519f8963ed2a762c57506e1d07984b0fb887/tvspy-backend/src/routes/subcriptions.js#L34](https://github.com/alebupal/tvspy/blob/53b7519f8963ed2a762c57506e1d07984b0fb887/tvspy-backend/src/routes/subcriptions.js#L34)) write `console.log(rawMessage)`
5. Press esc and type to save and exit :wq
6. `exit `to exit the container
7. `docker restart tvspy`

And now pass me what appears in the docker log when you start a reproduction

thank you!

**Comment by kapiszon53** [comment]
So this is the output, fresh after container restart and starting one reproduction:

https://pastebin.com/pHrdq9Gq

**Comment by alebupal** [comment]
I am seeing that the connection to the socket is correct. You have data in the "Registry" option, right?

**Comment by kapiszon53** [comment]
Yes, everything else seems to be working well. Just no active plays
![image](https://github.com/user-attachments/assets/d8b74060-876b-4f87-b984-240a51daac5f)
![image](https://github.com/user-attachments/assets/97c35106-aef6-4a41-8e86-830eed29dba2)

**Comment by alebupal** [comment]
Ok, so it's a front-end problem. What browser do you use?

If you open the element inspector (right click on the web, inspect) you get this comment?
`WebSocket - Conectado al servidor`
![image](https://github.com/user-attachments/assets/0fb23c76-9264-4521-8a36-95e58d0fefa4)

**Comment by alebupal** [comment]
Ok, I'm going to prepare a new version in the next few days so I can activate a debug mode to see the trace of where it fails. In the frontend it's not so easy to modify files and see the error.

**Comment by kapiszon53** [comment]
Looking forward to that, thanks.
Let me know if you need more information and I'll provide

**Comment by alebupal** [comment]
I have added the debug mode in the :dev tag, I still need to test a few things but you can already see the trace in the inspector to see if we can find the problem.

**Comment by alebupal** [comment]
I have added debug mode to the latest version

**Comment by kapiszon53** [comment]
Console output from inspector:
`content-script.js:104 Failed to get subsystem status for purpose {rejected: true, message: 'UNSUPPORTED_OS'}
index-CGERL90E.js:72 WebSocket - Conexión cerrada`

**Comment by alebupal** [comment]
Can you click on content-script.js:104 and show me a screenshot of this script, it doesn't sound familiar to me

**Comment by kapiszon53** [comment]
After a while, when jumping over the channels, this shows up. Not sure if it's related to my problem
![image](https://github.com/user-attachments/assets/7c2e8d01-a36d-4e9e-81da-91210c77a735)

**Comment by alebupal** [comment]
runtime. lastError: A listener indicated an asynchronous response by returning true, but the message channel closed before a response was received

**Comment by alebupal** [comment]
> After a while, when jumping over the channels, this shows up. Not sure if it's related to my problem ![image](https://github.com/user-attachments/assets/7c2e8d01-a36d-4e9e-81da-91210c77a735)

Hi, I thought I was waiting for a message from you, but I saw that you didn't respond. Can you try the app in an incognito window to see if you still get that error message?

I have read that this error can be caused by various Chrome extensions.

**Comment by kapiszon53** [comment]
Hi, this time I tried with a fresh installation of Firefox to avoid conflicts with chrome extensions. Unfortunately, the same result occurs:

![Image](https://github.com/user-attachments/assets/9662f4e4-c651-4431-b18f-9f1a88c2e5db)

**Comment by kapiszon53** [comment]
If you mean the tvheadend server's port - it's 9981. I haven't changed that

**Comment by alebupal** [comment]
Ok, I've confused the tvspy port with the tvheadend port in the screenshot 🤦‍♂️
 I keep thinking, this is very strange

**Comment by alebupal** [comment]
Hello, during these vacation days I’ve been thinking that it might be due to issues with the encoding of the username and password. I’ve uploaded a new version to the **dev** tag to see if it works for you.

```
docker create  \
 --name=tvspy \
 -v <path to data>:/app/backend/src/database/file \
 -p xxxxx:80 \
 -e TZ=Europe/Madrid \
 -e LOCALE=es-ES \
 alebupal/tvspy:dev\
```
